### ISSUE TITLE
Change notification

### ISSUE THREAD

**Issue by zimt28** [Meteor-Reaction/Herald#61]
Is there a way to change notifications? So when I get two messages from a user, I'd prefer one notification that changes the message, than two separate notifications. A solution may be to create a new notification, and then delete the old one. Maybe I could save a counter on the notification, find the old one before deleting it and increment that counter on the new one. What do you think?

**Comment by Kestanous** [comment]
I was thinking you could use onRun but if this is for client notifications that would not work. You can however get at the herald collection with Herald.collection. Have you ever used [collection-hooks](https://github.com/matb33/meteor-collection-hooks/)? You could add a `.before.insert` hook and if you don't want to create second notification `return false`.

**Comment by zimt28** [comment]
Yes, I'm using collection-hooks in my app. I've now solved it like this:

```js
var Case = Cases.findOne(id);

if (Meteor.isServer) {
  var count = 1;

  var query = {
    'data.case._id': id,
    read: false
  };

  var oldNotification = Herald.collection.findOne(query);

  if (oldNotification) {
    count = oldNotification.data.count + 1;
    Herald.collection.update(query, { $set: { read: true }});
  }

  Herald.createNotification(Case.userId, {
    courier: 'newMessage',
    data: {
      case: {
        _id: Case._id,
        patient: Case.patient,
        xmlNumber: Case.xmlNumber
      },
      count: count
    }
  });
}
```

I'll see if I can find a nicer way to do this :)
